include/crm/services.h

```diff
@@ -392,41 +390,6 @@ gboolean services_alert_async(svc_action_t *action,
         }
     }
 
-    /**
-     * \brief Get OCF equivalent of LSB exit code
-     *
-     * \param[in] action        LSB action that produced exit code
-     * \param[in] lsb_exitcode  Exit code of LSB action
-     *
-     * \return PCMK_OCF_* constant that corresponds to LSB exit code
-     */
-    static inline enum ocf_exitcode
-    services_get_ocf_exitcode(const char *action, int lsb_exitcode)
-    {
-        /* For non-status actions, LSB and OCF share error code meaning <= 7 */
-        if (action && !pcmk__strcase_any_of(action, "status", "monitor", NULL)) {
```

Comment:
Changing a regular function to inline definitely breaks shared object compatibility, and I suspect inline-to-regular does too (if code compiled with the old header links against the new library, bad things could happen). We can just replace this call with strcasecmp().

Promoting pcmk__strcase_any_of() to public API would also be OK, but I don't think that's necessary for this one usage.

Reply:
Thousand possibilities ...
Removing as I did here is definitely no issue.
Making an inline into a regular function shouldn't break anything as long as they are doing the same stuff - previously compiled would just work as before and new compilation would require a new but backward compatible library.
Having stuff inline in external headers is just a pain if not really needed for performance-reasons as it doesn't just limit you to keeping the interface but as well the implementation as far as other headers are required.

Reply:
> Changing a regular function to inline definitely breaks shared object compatibility, and I suspect inline-to-regular does too (if code compiled with the old header links against the new library, bad things could happen). We can just replace this call with strcasecmp().

This one actually needs to just be strcmp.  There were two levels of patch that happened here.  First, convert to using pcmk__str_null_matches.  Second, rename pcmk__str_any_of to pcmk__strcase_any_of.  From the first patch:
```
-        if (action && strcmp(action, "status") && strcmp(action, "monitor")) {
+        if (action && !pcmk__str_any_of(action, "status", "monitor", NULL)) {
```
So I guess now I have something else to go check for if I broke.

Reply:
> Making an inline into a regular function shouldn't break anything as long as they are doing the same stuff - previously compiled would just work as before and new compilation would require a new but backward compatible library.

Are you sure? Compilers aren't required to respect "inline", so couldn't there be a symbol conflict if they ignore it?

> Having stuff inline in external headers is just a pain if not really needed for performance-reasons as it doesn't just limit you to keeping the interface but as well the implementation as far as other headers are required.

Agreed, we overuse inline

Reply:
> 
> Are you sure? Compilers aren't required to respect "inline", so couldn't there be a symbol conflict if they ignore it?
> 
Exactly which is why I would in this case use a new name for the inline function and a macro to map the old name to the new one. Like this it shouldn't be a problem to in the end have both symbols if inlining doesn't happen for whatever reason.